### ISSUE TITLE
Kind request for pylint rule E0110

### ISSUE THREAD

**Issue by shager** [astral-sh/ruff#6806]
Hello,

I would like to ask whether it would be possible for one of the ruff developers to implement the pylint rule "E0110" (abstract-class-instantiated).
Here is an example for a code that produces a finding with pylint:
```python
import abc


class A(abc.ABC):
    @abc.abstractmethod
    def f(self):
        """abstract"""


class B(A):
    pass

    @abc.abstractmethod
    def f(self):
        """still abstract"""


def main():
    b = B()     # <--------- there should be a finding here 


if __name__ == "__main__":
    main()
```

The relevant pylint output is:
```
...
blub.py:19:8: E0110: Abstract class 'B' with abstract methods instantiated (abstract-class-instantiated)
...
```

Thank you very much in advance!

**Comment by charliermarsh** [comment]
Thank you for the kind issue :)

I'm going to merge this into the broader Pylint tracking issue #970. We can definitely implement this, but until we support multi-file analysis (analyzing code _across_ files), we'd only be able to detect this for instantiations of classes in the file in which they're defined.

**Comment by ChaserZ98** [comment]
New Ruff user here. Is there any plan of implementing this rule and abstract-method W0223 this year? It would help a lot in developing OO projects.
